cli/flox/src/commands/auth.rs

```diff
@@ -74,25 +76,101 @@ pub async fn authorize(client: BasicClient) -> Result<Credential> {
 
     debug!("Device code details: {details:#?}");
 
-    eprintln!(
-        "Please visit {} in your browser",
-        details.verification_uri().as_str()
+    let url = details.user_code().secret().clone();
+    let prefilled_url = details
+        .verification_uri_complete()
+        .map(|uri| uri.secret())
+        .cloned();
+
+    eprintdoc! {"
+        First copy your one-time code: {code}
+
+        Press [enter] to open {url} in your browser.
+        ",
+        url = details.verification_uri().as_str(),
+        code = details.user_code().secret()
+    };
+
+    debug!(
+        "Waiting for user to enter code (timeout: {}s)",
+        details.expires_in().as_secs()
     );
-    eprintln!("And enter code: {}", details.user_code().secret());
 
-    let token_result = client
+    // in he background listen for `[enter]` key presses
+    // if the user presses enter, open the browser using the system default opener
+    // on linux this should be `xdg-open`
+    // on macos this should be `open`
+    //
+    // I'm not sure this is the best way to do this, but it works for now.
```

Comment:
Question: would it be possible to just use `Dialog` rather than spawning an async task? This feels overly complicated.

Reply:
There's polling below that checks if the user authenticates without ever pressing enter. How would you poll both for user input and for authentication at the same time without doing something async?

Reply:
polling is gone now - per design